src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java

```diff
@@ -0,0 +1,611 @@
+			throw new InterceptorException("Couldn't find all required OGC " +
+					"parameters (SERVICE, REQUEST, LAYER). Please check the " +
+					"validity of the request.");
+		}
+
+		return ogcRequest;
+	}
+
+	/**
+	 *
+	 * @param requestService
+	 * @param requestOperation
+	 * @param requestLayer
+	 * @return
+	 */
+	private InterceptorRule getMostSpecificRule(String requestService,
+			String requestOperation, String requestEndPoint, String ruleEvent) {
+
+		List<InterceptorRule> result;
+
+		Map<String, String> filterMap = new HashMap<String, String>();
+
+		filterMap.put("event", ruleEvent);
+
+		filterMap.put("service", requestService);
+		filterMap.put("operation", requestOperation);
+		filterMap.put("endPoint", requestEndPoint);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service, Operation and EndPoint " +
+					"set in interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service and EndPoint set in " +
+					"interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", requestOperation);
+		filterMap.put("endPoint", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service and Operation set in " +
+					"interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", null);
+		filterMap.put("endPoint", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service set in interceptor rule");
+			return result.get(0);
+		}
+
+		return null;
+
+	}
+
+	/**
+	 *
+	 * @param mutableRequest
+	 * @param key
+	 * @return
+	 * @throws InterceptorException
+	 */
+	private static String getRequestParameterValue(MutableHttpServletRequest mutableRequest,
+			String[] keys) throws InterceptorException {
+
+		String value = StringUtils.EMPTY;
+
+		for (String key : keys) {
+			value = getRequestParameterValue(mutableRequest, key);
+
+			if (StringUtils.isNoneEmpty(value)) {
+				break;
+			}
+		}
+
+		return value;
+	}
+
+	/**
+	 *
+	 * @param mutableRequest
+	 * @param key
+	 * @return
+	 * @throws InterceptorException
+	 */
+	private static String getRequestParameterValue(MutableHttpServletRequest mutableRequest,
+			String key) throws InterceptorException {
+
+		if (StringUtils.isEmpty(key)) {
+			throw new InterceptorException("Missing parameter key");
+		}
+
+		String value = StringUtils.EMPTY;
+
+		Map<String, String[]> queryParams = mutableRequest.getParameterMap();
+
+		if (!queryParams.isEmpty()) {
+
+			TreeMap<String, String[]> params = new TreeMap<String, String[]>(
+					String.CASE_INSENSITIVE_ORDER);
+
+			params.putAll(queryParams);
+
```

Comment:
To be case insensitive while searching for a request parameter, we have to use a `TreeMap` with constructor `String.CASE_INSENSITIVE_ORDER`.